tests/python/test_python_frontend.py

```diff
@@ -4432,6 +4432,28 @@ def fusion_func(fd: FusionDefinition):
 
         nvf_out, _ = self.exec_nvfuser(fusion_func, inputs)
 
+    def test_returning_aliased_outputs(self):
+        inputs = [
+            torch.randn((1, 2, 3, 4), dtype=torch.float32, device="cuda:0")
+        ]
+
+        def fusion_func(fd: FusionDefinition):
+            T0 = fd.define_tensor(shape=[-1, -1, -1, -1], contiguity=[True, True, True, True], dtype=DataType.Float, is_cpu=False, stride_order=[3, 2, 1, 0])
+            S1 = fd.define_scalar(0.00000, dtype=DataType.Double)
+            T2 = fd.ops.gt(T0, S1)
+            S3 = fd.define_scalar(0.00000, dtype=DataType.Double)
+            T4 = fd.ops.where(T2, T0, S3)
+            fd.add_output(T4)
+            fd.add_output(T4, T0)
```

Comment:
I think our python API looks bad here (I'm the one to be blamed).

We should have really renamed this to `aliased_update` or something like that.

We can revisit this later. For the time being, this fixes the thunder issue that @crcrpar hit in https://github.com/Lightning-AI/lightning-thunder/pull/749